### ISSUE TITLE
Something is wrong with --detect-shared-borders and -S

### ISSUE THREAD

**Issue by ericfischer** [mapbox/tippecanoe#482]
```
tippecanoe -D10 -S5 --detect-shared-borders -X -f -o foo.mbtiles tl_2010_06001_tabblock10.json
```

has visible big holes at z9, and even worse gaps at z8 and z7.

```
tippecanoe -D10 --detect-shared-borders -X -f -o foo.mbtiles tl_2010_06001_tabblock10.json
```

looks OK without the `-S`.

**Comment by ericfischer** [comment]
It looks like this is actually a problem with tiny polygon reduction:

```
tippecanoe --no-tiny-polygon-reduction -D10 -S5 -z7 --detect-shared-borders -X -f -o foo.mbtiles tl_2010_06001_tabblock10.json
```

does not have the gaps (but does seem to have a few overlapping blocks and blocks with cut corners).

**Comment by ericfischer** [comment]
Most of the overlapping blocks in this case are "revived" polygons from cases where quantization to the tile grid reduced a polygon to a line or inverted its area.

**Comment by ericfischer** [comment]
The same number of tiny polygons are reduced with and without `--detect-shared-borders`, so maybe the reduction is just hidden by being surrounded by lots of badly-simplified blocks?

**Comment by ericfischer** [comment]
No, there are ~4000 features (out of 15000) that appear in

```
tippecanoe -D10 -S5 -Z7 -z7 -X -f -o foo.mbtiles tl_2010_06001_tabblock10.json
```

but not in

```
tippecanoe -D10 -S5 -Z7 -z7 --detect-shared-borders -X -f -o foo.mbtiles tl_2010_06001_tabblock10.json
```

What are they, and why are they being lost?

**Comment by ericfischer** [comment]
The missing features are coming out of `find_common_edges` with only 3 points in the geometry (a doubled-back line, not a valid polygon).

**Comment by ericfischer** [comment]
Almost everything that is coming back as just 3 points was represented as 2 arcs in the arc stage.

Arc simplification is calling `simplify_lines` with a parameter to keep at least 3 points per arc, but either they started out with fewer or it's not working.

**Comment by ericfischer** [comment]
Maybe the arcs are so short because the coordinates are being pre-quantized to z7 during input, and it needs to turn quantization off if it is going to detect shared borders.

**Comment by ericfischer** [comment]
Fixed in ecbfad2 by raising the minimum number of points in each simplified arc.

**Comment by ericfischer** [comment]
Merged
